flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/AggSqlFunction.scala

```diff
@@ -57,6 +57,8 @@ class AggSqlFunction(
   ) {
 
   def getFunction: AggregateFunction[_, _] = aggregateFunction
+
+  override def isDeterministic: Boolean = aggregateFunction.isDeterministic
```

Comment:
`isDeterministic` is the base method of `SqlOperator` which is the base class of `AggSqlFunction`, `ScalarSqlFunction`, `TableSqlFunction`. So I think maybe it's not necessary to add an abstract class.